### PR TITLE
fix: fix secrets docs by adding resource creation and permissions

### PR DESCRIPTION
**Pull request by jyecusch**

### CODE REVIEW

File: docs/reference/secrets/secret-put.mdx [hunk 1]
```diff
@@ -39,9 +39,9 @@ await keyRef.put('6c3199a3-094e-4797-bfc9-9ee2a7839286');
 Calling `put()` returns a promise to a reference to the new secret version. Storing the ID of the new version can be useful if you need to retrieve that specific value again in future using [version.access()](./secret-version-access)
```
Comment (raksiv) [inline docs/reference/secrets/secret-put.mdx:39]: For consistency change to - version().access()

Comment (HomelessDinosaur) [reply docs/reference/secrets/secret-put.mdx:39]: The other instance references it as: `secret.version().access()`

Comment (jyecusch) [reply docs/reference/secrets/secret-put.mdx:39]: It's context-dependent, in this context we're saying they already have a reference to a `version` object (returned from `put()`) so they can call `access()` on that object. In other examples the version reference doesn't exist yet, so we're saying get the version, then call access i.e. `secret.version().access()` or `secret.latest().access()`.

Comment (jyecusch) [reply docs/reference/secrets/secret-put.mdx:39]: Maybe it's confusing to overload the object reference and method names, but that would be a bigger change to fix.

Comment (raksiv) [reply docs/reference/secrets/secret-put.mdx:39]: Ah I see, a little confusing, but since it doesn't degrade the user experience let's leave it for now.

**Review by raksiv — APPROVED** [review]

**Review by asalkeld — APPROVED** [review]

**Pull request merged**